src/main/java/net/digibi/mailerlite/MailerLiteClient.java

```diff
@@ -12,34 +15,92 @@
+        HttpResponse<String> response = Unirest.post(API_END_POINT + "groups/" + groupId + "/subscribers").header("content-type", "application/json").header("x-mailerlite-apikey", API_KEY)
+                .body(userBody).asString();
         return response.isSuccess();
     }
 
-    public boolean subscribeUser(long groupId, String email, String name) {
-        return subscribeUser(groupId, createUserBodyToSend(email, name, null));
+    /**
+     * Moves user(identified by email, name) to specified groupId and set type of subscription
+     * @param groupId
+     * @param email
+     * @param type allowed values: unsubscribed, active, unconfirmed
+     * @return
+     */
+    public boolean moveSubscriberToGroup(long groupId, String email, String type) {
+        return moveSubscriberToGroup(groupId, createUserBodyToSend(email, null, type));
     }
 
-    public boolean subscribeUser(long groupId, String email, String name, String fields) {
-        return subscribeUser(groupId, createUserBodyToSend(email, name, fields));
+    /**
+     * Moves user(identified by email, name) to specified groupId, set fields & type of subscription
+     * @param groupId
+     * @param email
+     * @param fields fields to set
+     * @param type allowed values: unsubscribed, active, unconfirmed
+     * @return
+     */
+    public boolean moveSubscriberToGroup(long groupId, String email, Map<String, String> fields, String type) {
+        return moveSubscriberToGroup(groupId, createUserBodyToSend(email, fields, type));
     }
 
-    public static String createUserBodyToSend(String email, String name, String fields) {
-        JSONObject jsonObject = new JSONObject().put("email", email).put("name", name);
-        if (fields!=null) {
-            jsonObject.put("fields", fields);
+    /**
+     * Method to construct String in JSON format
+     * 
+     * @param email
+     * @param fields
+     * @param type   values accepted: unsubscribed, active
+     * @return String in JSON format
+     */
+    public static String createUserBodyToSend(String email, Map<String, String> fields, String type) {
+        JSONObject jsonObject = new JSONObject().put("email", email).put("type", type);
+        if (fields != null) {
+            jsonObject.put("fields", fields.toString());
         }
         return jsonObject.toString();
     }
+    
+    public HttpResponse<String> updateUserDetails(int identifier, Map<String, String> fields) {
+        JSONObject fieldsParsed = new JSONObject().put("fields", new JSONObject(fields));
+        HttpResponse<String> response = Unirest
+            .put(API_END_POINT + "subscribers/" + identifier)
+            .header("content-type", "application/json")
+            .header("x-mailerlite-apikey", API_KEY)
+            .body(fieldsParsed.toString())
+            .asString();
+        return response;
+    }
+    
+    public HttpResponse<String> updateUserDetails(String identifier, Map<String, String> fields) {
```

Comment:
use `email` instead of `identifier`

Reply:
The code is duplicated with the previous method. Maybe you can exact them to a single private method and use these methods as our API endpoint.